### ISSUE TITLE
Compilation Error with Stockfish 15.1 on Apple Silicon-based MacBook Pro using Alpine 3.18.0 in Docker

### ISSUE THREAD

**Issue by ricardobalk** [official-stockfish/Stockfish#4580]
### Describe the issue

When attempting to compile Stockfish 15.1 on an Apple Silicon-based MacBook Pro using Docker, an error occurs due to the `-fexperimental-new-pass-manager` flag, which is no longer supported in Clang 16.

### Expected behavior

Stockfish 15.1 should compile successfully on the Apple Silicon-based MacBook Pro using Alpine 3.18.0 running in Docker.

### Steps to reproduce

On an Apple Silicon-based MacBook Pro with a working installation of Docker, run the following commands in a terminal, to reproduce the problem:

```bash
docker run --rm -it alpine:3.18.0
apk add --no-cache git make llvm clang libc-dev
git clone --depth=1 https://github.com/official-stockfish/Stockfish.git stockfish
cd $_ # or 'cd stockfish'
git fetch --tags --prune
git checkout sf_15.1 # HEAD is now at 758f9c93 Stockfish 15.1
cd ./src/
make -j build ARCH=apple-silicon COMP=clang
```

Will give the following error:
```
clang-16: error: unknown argument: '-fexperimental-new-pass-manager'
```

### Anything else?

The issue is caused by the `-fexperimental-new-pass-manager` flag, which is present in the Makefile of Stockfish 15.1. However, this flag has been removed in Clang 16.

https://github.com/official-stockfish/Stockfish/blob/758f9c9350abee36a5865ec701560db8ea62004d/src/Makefile#L574

The new pass manager has been the default since Clang 13. Starting from Clang 16, the `-fexperimental-new-pass-manager` flag is no longer necessary. You can refer to the  [Clang 16.0.0 Release Notes](https://releases.llvm.org/16.0.0/tools/clang/docs/ReleaseNotes.html) for more information. Also, see https://github.com/llvm/llvm-project/commit/69b2b7282e92a1b576b7bd26f3b16716a5027e8e.

My current solution to resolve the issue is by commenting out the flag in the Makefile. You can use the following command to comment out the line in the Makefile: `sed -i '574s/^/#/' Makefile`. After commenting out the flag, Stockfish 15.1 compiles successfully. 🎉 (with some warnings about unused arguments, but that's unrelated to this issue)

```bash
docker run --rm -it alpine:3.18.0
apk add --no-cache git make llvm clang libc-dev
git clone --depth=1 https://github.com/official-stockfish/Stockfish.git stockfish
cd $_ # or 'cd stockfish'
git fetch --tags --prune
git checkout sf_15.1 # HEAD is now at 758f9c93 Stockfish 15.1
cd ./src/
sed -i '574s/^/#/' Makefile # comments out the flag -fexperimental-new-pass-manager
make -j build ARCH=apple-silicon COMP=clang
```

I think we should consider implementing a Clang version check in the Stockfish build process. By checking the Clang compiler version and determining if the version is less than 13, we can include the `-fexperimental-new-pass-manager` conditionally. This would ensure compatibility with different Clang versions and avoid compilation errors like the one encountered with Clang 16.

Another option would be to remove the `-fexperimental-new-pass-manager` flag entirely from the Makefile. However, this could mean that users compiling Stockfish on older versions of Clang (before version 13) may experience compatibility issues or potential performance impacts.

Therefore, implementing a Clang version check would be my preferred solution.

### Operating system

MacOS

### Stockfish version

758f9c9350abee36a5865ec701560db8ea62004d

**Comment by MinetaS** [comment]
Hi, this is already fixed in d1e17989b51f220629e4e81a4bf413974f4b18e2. Please try with the latest version and see if the code is compiled correctly.

**Comment by vondele** [comment]
This fix will be part of a future release, and is already part of our current development branch. Closing as resolved.
